MoviesStoreApp/MVPMovies/MoviesListPresenter/MoviesListPresenter.swift

```diff
@@ -0,0 +1,62 @@
+// MoviesListPresenter.swift
+// Copyright © RoadMap. All rights reserved.
+
+/// Презентер
+final class MoviesListPresenter: MoviesListPresenterProtocol {
+    // MARK: - Public Properties
+
+    var movies: [Movie]?
+
+    // MARK: - Private Properties
+
+    private var view: MoviesListViewProtocol?
+    private var routerProtocol: RouterProtocol?
+    private let networkService: NetworkServiceProtocol?
+
+    // MARK: - Init
+
+    required init(
+        view: MoviesListViewProtocol,
+        networkService: NetworkServiceProtocol,
+        routerProtocol: RouterProtocol
+    ) {
+        self.view = view
+        self.networkService = networkService
+        self.routerProtocol = routerProtocol
+        fetchMovies(movieType: URLRequest.popularRequest)
+    }
+
+    // MARK: - Public Methods
+
+    func getTypeMovies(index: Int) {
+        switch index {
+        case 0:
+            fetchMovies(movieType: URLRequest.topRatedRequest)
+        case 1:
+            fetchMovies(movieType: URLRequest.popularRequest)
+        case 2:
+            fetchMovies(movieType: URLRequest.upcomingRequest)
+        default:
+            break
+        }
+    }
+
+    // MARK: - Private Methods
+
+    private func fetchMovies(movieType: String) {
+        networkService?.fetchMovies(moviesType: movieType) { [weak self] response in
+            guard let self = self else { return }
+            switch response {
+            case let .success(movie):
+                self.movies = movie
+                self.view?.success()
+            case let .failure(error):
+                self.view?.failure(error: error)
+            }
+        }
+    }
+
+    func showMovieInfo(movieID: String) {
```

Comment:
```suggestion
    func showMovieInfo(movieID: String) {
```
 поднять над приватными